src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java

```diff
@@ -130,23 +124,49 @@ public static void logOnLowLevel(String command, String description, boolean suc
   }
 
   private static void log(String command, String description, boolean isSuccess,
-      boolean ifLowLevel) {
+                          boolean ifLowLevel) {
     logsResults.add(isSuccess);
     String escapedDescription = escapeHtml(description);
 
-    String className = isSuccess ? "success" : "error";
-    StringBuilder builder = new StringBuilder();
+    String className = isSuccess ? SUCCESS_CLASS : ERROR_CLASS;
+    List<String> classList = new ArrayList<>();
+    classList.add(className);
+
     if (ifLowLevel) {
-      builder.append("<tr class=\"" + className + " lowLevelAction" + "\"><td>" + command
-          + "</td><td>" + escapedDescription + "</td><td> <br/> &nbsp;</td></tr>");
-    } else {
-      builder.append("<tr class=\"" + className + "\"><td>" + command + "</td><td>"
-          + escapedDescription + "</td><td> <br/> &nbsp;</td></tr>");
+      classList.add(LOW_LEVEL_ACTION_CLASS);
     }
-    CommonUtils.appendTextToFile(logPath, builder.toString());
+    String html = VelocityWrapper.fillLogRow(classList, command, escapedDescription);
+    CommonUtils.appendTextToFile(logPath, html);
     logJSError(DriverProvider.getActiveDriver());
   }
 
+  public void logAssertionStacktrace(AssertionError exception) {
+    driver = DriverProvider.getActiveDriver();
+
+    imageCounter += 1;
+    if ("true".equals(Configuration.getLogEnabled())) {
+      try {
+        new Shooter().savePageScreenshot(screenPath + imageCounter, driver);
+        CommonUtils.appendTextToFile(screenPath + imageCounter + ".html", getPageSource(driver));
+      } catch (Exception e) {
+        log("onException",
+            "driver has no ability to catch screenshot or html source - driver may died", false);
+      }
+      String exceptionMessage = ExceptionUtils.getStackTrace(exception);
+
+      //TODO: Check imageCounter
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1135)